src/mlpack/tests/customlayer.hpp

```diff
@@ -0,0 +1,24 @@
+#ifndef CUSTOMLAYER_HPP
+#define CUSTOMLAYER_HPP
```

Comment:
Just to match the rest of the mlpack style, I'd consider naming this `custom_layer.hpp` and use the header guard `MLPACK_TESTS_CUSTOM_LAYER_HPP`.  Also, we should add a comment at the start of the file with the filename, what it contains, who wrote it, etc., and the license.  You can see any of the other files in mlpack for an idea. :)